src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/db/entities/ProbationDeliveryUnit.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.hmpps.hmppsprobationestateapi.db.entities
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.Transient
+import org.springframework.data.domain.Persistable
+import org.springframework.data.relational.core.mapping.Table
+import java.time.ZonedDateTime
+import java.time.temporal.ChronoUnit
+
+@Table("pdu")
+data class ProbationDeliveryUnit constructor(
+  @Id val code: String,
+  val name: String,
+  val createdDate: ZonedDateTime = ZonedDateTime.now().truncatedTo(ChronoUnit.SECONDS),
+  val regionCode: String,
+  @Transient @Value("false") val new: Boolean
```

Comment:
Added this because the error `org.springframework.dao.TransientDataAccessResourceException: Failed to update table [region]. Row with Id [REGION1] does not exist.` is thrown.

Two options here:

1. Add a generated ID column and don't have the code column as the ID, if null its considered a new entry to generates an INSERT, if set it would be an update. Would need to make sure only 1 entry with the code value is stored
2.  Implement Persistable (what is currently in this PR) and add a Transient boolean which is set to true if its new and false from the DB